src/developer/core/plugin-api.md

```diff
@@ -69,16 +72,120 @@ This is the workhorse.
 * Records video and data.
 Most, and preferably all coordination and control happens within the World process.
 
+### Player Process
+Main process within Pupil Player.
+
+* Visualizes recorded data
+* Performs post-hoc analyses
+* Exports video and csv data
+
 ## API Reference
 
-All plugins are required to be subclasses from the root [Plugin class](https://github.com/pupil-labs/pupil/blob/master/pupil_src/shared_modules/plugin.py#L25-L167). It is available during runtime:
+Plugins are the recommended way to extend Pupil Core software functionality. They are
+typically managed within the application's main process which communicates with the
+plugins via callbacks (see below). These are defined in the root
+[`Plugin` class](https://github.com/pupil-labs/pupil/blob/master/pupil_src/shared_modules/plugin.py).
+In order to create a new plugin, one has to inherit from this class and overwrite them
+as desired.
+
 ```py
 from plugin import Plugin
 
 class MyCustomPlugin(Plugin):
     pass
 ```
 
+Plugins are automatically listed in the [Plugin Manager](/core/software/pupil-capture/#plugins)
+unless they inherit from a set of special classes (`System_Plugin_Base`, `Base_Manager`,
+`Base_Source`, `CalibrationChoreographyPlugin`, `GazerBase`).
+
+### Plugin class attributes
+
+These class attributes define general plugin behavior. Overwriting them is optional but
+recommneded. `alive` is an exception and should only be set to `False` if you want the
+plugin to close autonomously. Otherwise, this attribute is managed by the enclosing
+application.
+
+| Name         | Possible values                                                                                        | Default value | Meaning                                                              |
+|--------------|--------------------------------------------------------------------------------------------------------|---------------|----------------------------------------------------------------------|
+| `uniqueness` | `"not_unique"`, `"by_class"`, `"by_base_class"`                                                        | `"by_class"`  | Plugin instance replacement behavior                                 |
```

Comment:
I think uniqueness would benefit from a slightly more detailed explanation of 'Meaning'